### DISCUSSION TITLE
Errors after Next.12 upgrade

### DISCUSSION THREAD

**Discussion by iamchathu** [vercel/next.js#30519] [Help]
### What version of Next.js are you using?

12.0.1

### What version of Node.js are you using?

14.18.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next export

### Describe the Bug

```bash
$ next -p 7777
ready - started server on 0.0.0.0:7777, url: http://localhost:7777
info  - Loaded env from /Users/Chathu/Workspace/GeniusSoft/pos-v4/pos-v4-ui/packages/buddhi-batiks/.env
warn  - The `target` config is deprecated and will be removed in a future version.
See more info here https://nextjs.org/docs/messages/deprecated-target-config
Error: Cannot find module './blocks/images'
Require stack:
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/build/webpack/config/index.js
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/build/webpack-config.js
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/server/dev/hot-reloader.js
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/server/dev/next-dev-server.js
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/server/next.js
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/server/lib/start-server.js
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/cli/next-dev.js
- /Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.mod._resolveFilename (/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/build/webpack/require-hook.js:171:28)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/build/webpack/config/index.js:8:15)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/build/webpack/config/index.js',
    '/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/build/webpack-config.js',
    '/Users/Chathu/Workspace/prrojectui/node_modules/next/dist/server/dev/hot-reloader.js',
    '/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/server/dev/next-dev-server.js',
    '/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/server/next.js',
    '/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/server/lib/start-server.js',
    '/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/cli/next-dev.js',
    '/Users/Chathu/Workspace/prroject/ui/node_modules/next/dist/bin/next'
  ]
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Expected Behavior

Run and build withour issues

### To Reproduce

Upgrade to Next.js 12 in @mui/material 5 project, in lerna based yarn workspaces mono-repo

**Comment by timneutkens** [comment]
It seems that your package manager (npm/yarn/pnpm) did not correctly download all files, the file that is reported missing is definitely published to the npm registry. Can you try removing `node_modules` and re-installing?

**Reply by iamchathu** [reply]
I'm using yarn 1.22.15

**Reply by iamchathu** [reply]
@timneutkens I have yarn autoclean setup. Does it has something to do with is issue?

**Reply by iamchathu** [reply]
I did a clean install yet the error is the same.

**Reply by iammarkps** [reply]
I also did the clean install and got the same error.

**Comment by imranbarbhuiya** [comment]
Try deleting node_modules and yarn.lock files and do `yarn install`.

**Comment by bryanrsmith** [comment]
I tracked this down to a `.yarnclean` file, which supplies yarn with patterns to clean from node_modules after an install. Yarn has a command that generates this file with a bunch of default patterns, one of which is `images`. The intent, I suppose, being to strip images from node_modules, making things like build caches faster. next@12 added a directory named `images` to its npm package, which is matching this rule and getting removed by yarn after the install completes.

You can either remove the `images` pattern from `.yarnclean`, or remove the `.yarnclean` file to disable this feature of yarn. Then delete node_modules and run `yarn` again, and you should be good.

**Reply by iamchathu** [reply]
Yeah, this is the case. @timneutkens I think best to respect default `.yarnclean` file. What do you think?

**Reply by HectorLS** [reply]
Thank you so much @bryanrsmith !

**Reply by buraksakalli** [reply]
Thanks! It works 🎉

**Reply by dipakparmar** [reply]
Thanks, @bryanrsmith, I got the same issue, fixed by commenting out the images in `.yarnclean`.

**Reply by LeonardoRomualdo** [reply]
Thanks man, it worked fine!

**Reply by aquiseb** [reply]
Thanks to your comment I realised that in my case, [node-prune](https://github.com/tj/node-prune/tree/master) was responsible upon Heroku deployment, I cloned the library and commented out images, then built it for linux and deployed again running the bin manually, it worked.

**Comment by Praneeth-Pike** [comment]
I have the same issue. Tried clean install but no use.

```
Error: Cannot find module './blocks/images'
Require stack:
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/build/webpack/config/index.js
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/build/webpack-config.js
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/dev/hot-reloader.js
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/dev/next-dev-server.js
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/next.js
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/lib/start-server.js
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/cli/next-dev.js
- /Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/bin/next
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.mod._resolveFilename (/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/build/webpack/require-hook.js:171:28)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/build/webpack/config/index.js:8:15)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/build/webpack/config/index.js',
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/build/webpack-config.js',
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/dev/hot-reloader.js',
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/dev/next-dev-server.js',
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/next.js',
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/server/lib/start-server.js',
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/cli/next-dev.js',
    '/Users/praneethpike/Work/projects/podium/podium-cms/node_modules/next/dist/bin/next'
  ]
}
```

**Reply by Praneeth-Pike** [reply]
I've tried what @bryanrsmith had suggested and it works now!

**Comment by code-by-sia** [comment]
I had same issue after switching the package manager from npm to yarn; That's how I fixed it.
First, I deleted the build folder (`.next`) as well as the dependencies folder `node_modules`; Afterwards, I install and run the project, and it worked.

**Comment by guojianlan** [comment]
I have the same issue. Tried clean install but no use. It usually throws the error ”Critical dependency: require function is used in a way in which dependencies cannot be statically extracted“ in my custom 3rd。

**Reply by guojianlan** [reply]
This error will not be thrown in NextJs 10.

**Comment by Kosai106** [comment]
Currently experiencing this exact issue using Next 12.1.0 in a yarn workspace.

I've checked the `node_modules` folder and the file isn't there regardless of how many times I clear and reinstall.

<img width="210" alt="Screenshot 2022-03-02 at 16 45 02" src="https://user-images.githubusercontent.com/6379824/156396362-56a7e7de-45de-43c4-bf69-e197b1c4ffce.png">

Not really sure how to proceed from here but it's blocking my ability to work.

Edit: Only solution I've found so far has been to use Next 11 instead.

**Comment by vicocotea** [comment]
It works for me after deleting folder `.next` (+ `node_modules`, lock files)
